### PR TITLE
Refactor Azure logs to use common cloud logger

### LINKED ISSUE #16739: Refactor Azure module to use Logger tool and log relevant information
| Affected integration |
|---|
| Azure|

## Description

As described in #16717, To standardize the logging a refactor in this cloud module needs to be done to integrate the logging tool developed in the mentioned issue.

Logging refactor on the module should be done to log relevant information about the process.

Unit tests should be reviewed as well to adapt to the new logging method.

## Tasks
- [x] Test in a manager.
- [x] Test in an agent.
- [x] Unit tests without failures. Updated if there are any relevant changes.
- [x] Integration tests without failures. Updated if there are any relevant changes.
- [x] Update the documentation if necessary.
- [x] Add entry to the changelog if necessary.

### PR DESCRIPTION
**Pull request by EduLeon12**
|Related issue|
|---|
| Closes #16739 |

## Description

As explained in #16739, The Azure wodle has been updated to include the modularized cloud logger.

## Tests

All test related to the wodle has been performed [here](https://github.com/wazuh/wazuh/issues/16739#issuecomment-1864846884).

```python
(unit-test) eduardoleon@pop-os:~/git/wazuh(16739-refactor-azure-logs)$ pytest wodles/
=========================== test session starts ===========================
platform linux -- Python 3.9.16, pytest-7.4.2, pluggy-0.13.1
rootdir: /home/eduardoleon/git/wazuh
plugins: anyio-3.6.2, aiohttp-1.0.4, trio-0.7.0, html-3.0.0, asyncio-0.18.1, cov-3.0.0, metadata-2.0.2
asyncio: mode=auto
collected 852 items

wodles/aws/tests/test_aws_bucket.py ............................... [  3%]
................................................................... [ 11%]
................................................................... [ 19%]
............................................                        [ 24%]
wodles/aws/tests/test_aws_s3.py ..................                  [ 26%]
wodles/aws/tests/test_aws_service.py ....                           [ 27%]
wodles/aws/tests/test_cloudtrail.py ..                              [ 27%]
wodles/aws/tests/test_cloudwatchlogs.py ........................... [ 30%]
..........................                                          [ 33%]
wodles/aws/tests/test_config.py ................................... [ 37%]
...........................................                         [ 42%]
wodles/aws/tests/test_guardduty.py .................                [ 44%]
wodles/aws/tests/test_inspector.py ......                           [ 45%]
wodles/aws/tests/test_load_balancers.py ............                [ 46%]
wodles/aws/tests/test_s3_log_handler.py ....                        [ 47%]
wodles/aws/tests/test_server_access.py ............................ [ 50%]
.....                                                               [ 51%]
wodles/aws/tests/test_sqs_queue.py .......                          [ 51%]
wodles/aws/tests/test_tools.py ...............................      [ 55%]
wodles/aws/tests/test_umbrella.py ......                            [ 56%]
wodles/aws/tests/test_vpcflow.py ............................       [ 59%]
wodles/aws/tests/test_waf.py .......                                [ 60%]
wodles/aws/tests/test_wazuh_integration.py ........................ [ 63%]
................................................................... [ 71%]
...........                                                         [ 72%]
wodles/azure/tests/test_azure.py .................................. [ 76%]
................................................................... [ 84%]
............                                                        [ 85%]
wodles/azure/tests/test_orm.py ...............................      [ 89%]
wodles/docker-listener/tests/test_docker_listener.py .............. [ 90%]
.....                                                               [ 91%]
wodles/gcloud/tests/test_bucket.py ................................ [ 95%]
.                                                                   [ 95%]
wodles/gcloud/tests/test_gcloud.py .........                        [ 96%]
wodles/gcloud/tests/test_integration.py ........                    [ 97%]
wodles/gcloud/tests/test_subscriber.py ..............               [ 99%]
wodles/gcloud/tests/test_tools.py ........                          [100%]

============================ warnings summary =============================
../../.pyenv/versions/3.9.16/envs/unit-test/lib/python3.9/site-packages/pytest_aiohttp/plugin.py:28
  /home/eduardoleon/.pyenv/versions/3.9.16/envs/unit-test/lib/python3.9/site-packages/pytest_aiohttp/plugin.py:28: DeprecationWarning: The 'asyncio_mode' is 'legacy', switching to 'auto' for the sake of pytest-aiohttp backward compatibility. Please explicitly use 'asyncio_mode=strict' or 'asyncio_mode=auto' in pytest configuration file.
    config.issue_config_time_warning(LEGACY_MODE, stacklevel=2)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
===================== 852 passed, 1 warning in 3.08s ======================

```

### CODE REVIEW

**Comment by EduLeon12** [comment]
##  Check Update

The failure on the Unit test is expected since the `shared/` module that contains the logger has not been updated in the pipeline.

```command
```

File: wodles/azure/azure-logs.py [hunk 1]
```diff
@@ -195,16 +183,16 @@ def read_auth_file(auth_path: str, fields: tuple):
                     continue
                 credentials[key] = value.replace("\n", "")
         if fields[0] not in credentials or fields[1] not in credentials:
-            logging.error(f"Error: The authentication file does not contains the expected '{fields[0]}' "
+            azure_logger.error(f"Error: The authentication file does not contains the expected '{fields[0]}' "
```
Comment (fdalmaup) [inline wodles/azure/azure-logs.py:186]:
```suggestion
            azure_logger.error(f"Error: The authentication file does not contain the expected '{fields[0]}'."
```

File: wodles/azure/azure-logs.py [hunk 2]
```diff
@@ -195,16 +183,16 @@ def read_auth_file(auth_path: str, fields: tuple):
...
                           f"and '{fields[1]}' fields.")
             sys.exit(1)
         return credentials[fields[0]], credentials[fields[1]]
     except ValueError:
-        logging.error("Error: The authentication file format is not valid. "
+        azure_logger.error("Error: The authentication file format is not valid. "
```
Comment (fdalmaup) [inline wodles/azure/azure-logs.py:191]:
```suggestion
        azure_logger.error("Error: The authentication file format is not valid."
```

File: wodles/azure/azure-logs.py [hunk 3]
```diff
@@ -240,12 +228,12 @@ def update_row_object(table: orm.Base, md5_hash: str, new_min: str, new_max: str
     if new_min_date < old_min_date or new_max_date > old_max_date:
         min_ = new_min if new_min_date < old_min_date else old_min_str
         max_ = new_max if new_max_date > old_max_date else old_max_str
-        logging.debug(f"Attempting to update a {table.__tablename__} row object. "
+        azure_logger.debug(f"Attempting to update a {table.__tablename__} row object. "
```
Comment (fdalmaup) [inline wodles/azure/azure-logs.py:231]:
```suggestion
        azure_logger.debug(f"Attempting to update a {table.__tablename__} row object."
```

File: wodles/azure/azure-logs.py [hunk 4]
```diff
@@ -518,7 +506,7 @@ def build_graph_url(offset: str, md5_hash: str):
         if item is None:
             item = create_new_row(table=orm.Graph, query=args.graph_query, md5_hash=md5_hash, offset=offset)
     except orm.AzureORMError as e:
-        logging.error(f"Error trying to obtain row object from '{orm.Graph.__tablename__}' using md5='{md5}': {e}")
+        azure_logger.error(f"Error trying to obtain row object from '{orm.Graph.__tablename__}' using md5='{md5}': {e}")
```
Comment (fdalmaup) [inline wodles/azure/azure-logs.py:509]:
```suggestion
        azure_logger.error(f"Error trying to obtain row object from '{orm.Graph.__tablename__}' using md5='{md5}': {e}.")
```

File: wodles/azure/azure-logs.py [hunk 5]
```diff
@@ -578,18 +566,18 @@ def get_graph_events(url: str, headers: dict, md5_hash: str):
             if args.graph_tag:
                 value['azure_aad_tag'] = args.graph_tag
             json_result = dumps(value)
-            logging.info("Graph: Sending event by socket.")
+            azure_logger.info("Graph: Sending event by socket.")
             send_message(json_result)
 
         if len(values_json) == 0:
-            logging.info("Graph: There are no new results")
+            azure_logger.info("Graph: There are no new results")
```
Comment (fdalmaup) [inline wodles/azure/azure-logs.py:573]:
```suggestion
            azure_logger.info("Graph: There are no new results.")
```

File: wodles/azure/azure-logs.py [hunk 6]
```diff
@@ -652,15 +640,15 @@ def start_storage():
             if item is None:
                 item = create_new_row(table=orm.Storage, query=name, md5_hash=md5_hash, offset=offset)
         except orm.AzureORMError as e:
-            logging.error(f"Error trying to obtain row object from '{orm.Storage.__tablename__}' using md5='{md5}': {e}")
+            azure_logger.error(f"Error trying to obtain row object from '{orm.Storage.__tablename__}' using md5='{md5}': {e}")
```
Comment (fdalmaup) [inline wodles/azure/azure-logs.py:643]:
```suggestion
            azure_logger.error(f"Error trying to obtain row object from '{orm.Storage.__tablename__}' using md5='{md5}': {e}.")
```

File: wodles/azure/azure-logs.py [hunk 7]
```diff
@@ -652,15 +640,15 @@ def start_storage():
...
             sys.exit(1)
 
         min_datetime = parse(item.min_processed_date, fuzzy=True)
         max_datetime = parse(item.max_processed_date, fuzzy=True)
         desired_datetime = offset_to_datetime(offset) if offset else max_datetime
         get_blobs(container_name=container, prefix=args.prefix, blob_service=block_blob_service, md5_hash=md5_hash,
                   min_datetime=min_datetime, max_datetime=max_datetime, desired_datetime=desired_datetime)
-    logging.info("Storage: End")
+    azure_logger.info("Storage: End")
```
Comment (fdalmaup) [inline wodles/azure/azure-logs.py:651]:
```suggestion
    azure_logger.info("Storage: End.")
```

File: wodles/azure/azure-logs.py [hunk 8]
```diff
@@ -811,10 +799,10 @@ def get_token(client_id: str, secret: str, domain: str, scope: str):
             err_msg = f"The '{domain}' tenant domain was not found."
         else:
             err_msg = "Couldn't get the token for authentication."
-        logging.error(f"Error: {err_msg}")
+        azure_logger.error(f"Error: {err_msg}")
 
     except RequestException as e:
-        logging.error(f"Error: An error occurred while trying to obtain the authentication token: {e}")
+        azure_logger.error(f"Error: An error occurred while trying to obtain the authentication token: {e}")
```
Comment (fdalmaup) [inline wodles/azure/azure-logs.py:805]:
```suggestion
        azure_logger.error(f"Error: An error occurred while trying to obtain the authentication token: {e}.")
```

File: wodles/azure/azure-logs.py [hunk 9]
```diff
@@ -833,19 +821,19 @@ def send_message(message: str):
...
         if e.errno == 111:
-            logging.error("ERROR: Wazuh must be running.")
+            azure_logger.error("ERROR: Wazuh must be running.")
             sys.exit(1)
         elif e.errno == 90:
-            logging.error("ERROR: Message too long to send to Wazuh.  Skipping message...")
+            azure_logger.error("ERROR: Message too long to send to Wazuh.  Skipping message...")
         else:
-            logging.error(f"ERROR: Error sending message to wazuh: {e}")
+            azure_logger.error(f"ERROR: Error sending message to wazuh: {e}")
```
Comment (fdalmaup) [inline wodles/azure/azure-logs.py:836]:
```suggestion
            azure_logger.error(f"ERROR: Error sending message to wazuh: {e}.")
```

File: wodles/azure/azure-logs.py [hunk 10]
```diff
@@ -240,12 +228,12 @@ def update_row_object(table: orm.Base, md5_hash: str, new_min: str, new_max: str
...
         min_ = new_min if new_min_date < old_min_date else old_min_str
         max_ = new_max if new_max_date > old_max_date else old_max_str
-        logging.debug(f"Attempting to update a {table.__tablename__} row object. "
+        azure_logger.debug(f"Attempting to update a {table.__tablename__} row object. "
                       f"MD5: '{md5_hash}', min_date: '{min_}', max_date: '{max_}'")
         try:
             orm.update_row(table=table, md5=md5_hash, min_date=min_, max_date=max_, query=query)
         except orm.AzureORMError as e:
-            logging.error(f"Error updating row object from {table.__tablename__}: {e}")
+            azure_logger.error(f"Error updating row object from {table.__tablename__}: {e}")
```
Comment (fdalmaup) [inline wodles/azure/azure-logs.py:236]:
```suggestion
            azure_logger.error(f"Error updating row object from {table.__tablename__}: {e}.")
```

File: wodles/azure/azure-logs.py [hunk 11]
```diff
@@ -268,17 +256,17 @@ def create_new_row(table: orm.Base, md5_hash: str, query: str, offset: str) -> o
...
     desired_str = desired_datetime.strftime(DATETIME_MASK)
     item = table(md5=md5_hash, query=query, min_processed_date=desired_str, max_processed_date=desired_str)
-    logging.debug(f"Attempting to insert row object into {table.__tablename__} with md5='{md5_hash}', "
+    azure_logger.debug(f"Attempting to insert row object into {table.__tablename__} with md5='{md5_hash}', "
                   f"min_date='{desired_str}', max_date='{desired_str}'")
     try:
         orm.add_row(row=item)
     except orm.AzureORMError as e:
-        logging.error(f"Error inserting row object into {table.__tablename__}: {e}")
+        azure_logger.error(f"Error inserting row object into {table.__tablename__}: {e}")
```
Comment (fdalmaup) [inline wodles/azure/azure-logs.py:269]:
```suggestion
        azure_logger.error(f"Error inserting row object into {table.__tablename__}: {e}.")
```

File: wodles/azure/azure-logs.py [hunk 12]
```diff
@@ -315,8 +303,8 @@ def start_log_analytics():
     try:
         get_log_analytics_events(url=url, body=body, headers=headers, md5_hash=md5_hash)
     except HTTPError as e:
-        logging.error(f"Log Analytics: {e}")
-    logging.info("Azure Log Analytics ending.")
+        azure_logger.error(f"Log Analytics: {e}")
```
Comment (fdalmaup) [inline wodles/azure/azure-logs.py:306]:
```suggestion
        azure_logger.error(f"Log Analytics: {e}.")
```

File: wodles/azure/azure-logs.py [hunk 13]
```diff
@@ -339,8 +327,8 @@ def build_log_analytics_query(offset: str, md5_hash: str) -> dict:
         if item is None:
             item = create_new_row(table=orm.LogAnalytics, query=args.la_query, md5_hash=md5_hash, offset=offset)
     except orm.AzureORMError as e:
-        logging.error(f"Error trying to obtain row object from '{orm.LogAnalytics.__tablename__}' using md5='{md5}': "
-                      f"{e}")
+        azure_logger.error(f"Error trying to obtain row object from '{orm.LogAnalytics.__tablename__}' "
+                           f"using md5='{md5}': {e}")
```
Comment (fdalmaup) [inline wodles/azure/azure-logs.py:331]:
```suggestion
                           f"using md5='{md5}': {e}.")
```

File: wodles/azure/azure-logs.py [hunk 14]
```diff
@@ -367,7 +355,7 @@ def build_log_analytics_query(offset: str, md5_hash: str) -> dict:
             filter_value = f"TimeGenerated > {max_str}"
 
     query = f"{args.la_query} | order by TimeGenerated asc | where {filter_value} "
-    logging.info(f"Log Analytics: The search starts for query: '{query}'")
+    azure_logger.info(f"Log Analytics: The search starts for query: '{query}'")
```
Comment (fdalmaup) [inline wodles/azure/azure-logs.py:358]:
```suggestion
    azure_logger.info(f"Log Analytics: The search starts for query: '{query}'.")
```

File: wodles/azure/azure-logs.py [hunk 15]
```diff
@@ -390,25 +378,25 @@ def get_log_analytics_events(url: str, body: dict, headers: dict, md5_hash: str)
...
-    logging.info("Log Analytics: Sending a request to the Log Analytics API.")
+    azure_logger.info("Log Analytics: Sending a request to the Log Analytics API.")
     response = get(url, params=body, headers=headers)
     if response.status_code == 200:
         try:
             columns = response.json()['tables'][0]['columns']
             rows = response.json()['tables'][0]['rows']
             if len(rows) == 0:
-                logging.info("Log Analytics: There are no new results")
+                azure_logger.info("Log Analytics: There are no new results")
```
Comment (fdalmaup) [inline wodles/azure/azure-logs.py:388]:
```suggestion
                azure_logger.info("Log Analytics: There are no new results.")
```

File: wodles/azure/azure-logs.py [hunk 16]
```diff
@@ -390,25 +378,25 @@ def get_log_analytics_events(url: str, body: dict, headers: dict, md5_hash: str)
...
             else:
                 time_position = get_time_position(columns)
                 if time_position is not None:
                     iter_log_analytics_events(columns, rows)
                     update_row_object(table=orm.LogAnalytics, md5_hash=md5_hash, new_min=rows[0][time_position],
                                       new_max=rows[len(rows) - 1][time_position], query=args.la_query)
                 else:
-                    logging.error("Error: No TimeGenerated field was found")
+                    azure_logger.error("Error: No TimeGenerated field was found")
```
Comment (fdalmaup) [inline wodles/azure/azure-logs.py:396]:
```suggestion
                    azure_logger.error("Error: No TimeGenerated field was found.")
```

File: wodles/azure/azure-logs.py [hunk 17]
```diff
@@ -465,37 +453,37 @@ def iter_log_analytics_events(columns: list, rows: list):
...
-    logging.info(f"Graph: Building the url.")
+    azure_logger.info(f"Graph: Building the url.")
     md5_hash = md5(args.graph_query.encode()).hexdigest()
     url = build_graph_url(offset=args.graph_time_offset, md5_hash=md5_hash)
-    logging.info(f"Graph: The URL is '{url}'")
+    azure_logger.info(f"Graph: The URL is '{url}'")
 
     # Get events
-    logging.info("Graph: Pagination starts")
+    azure_logger.info("Graph: Pagination starts")
```
Comment (fdalmaup) [inline wodles/azure/azure-logs.py:481]:
```suggestion
    azure_logger.info("Graph: Pagination starts.")
```

File: wodles/azure/azure-logs.py [hunk 18]
```diff
@@ -465,37 +453,37 @@ def iter_log_analytics_events(columns: list, rows: list):
...
     try:
         get_graph_events(url=url, headers=headers, md5_hash=md5_hash)
     except HTTPError as e:
-        logging.error(f"Graph: {e}")
-    logging.info("Graph: End")
+        azure_logger.error(f"Graph: {e}")
+    azure_logger.info("Graph: End")
```
Comment (fdalmaup) [inline wodles/azure/azure-logs.py:486]:
```suggestion
        azure_logger.error(f"Graph: {e}.")
    azure_logger.info("Graph: End.")
```

**Review by fdalmaup — CHANGES_REQUESTED** [review]

**Review by fdalmaup — APPROVED** [review]
LGTM!

**Review by Selutario — APPROVED** [review]

**Pull request merged**